### PR TITLE
Symantec Endpoint protection

### PR DESCRIPTION
**Pull request by squioc**
Add new format to parse events from Broadcom/Symantec Endpoint Protection

### CODE REVIEW

**Review by Darkheir — DISMISSED** [review]

File: Broadcom/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# BROADCOM
+
+## Description
+Broadcom is a technology company that supplies semiconductor and infrastructure software solutions
```
Comment (PierrickV) [inline Broadcom/README.md:4]:
```suggestion
Broadcom is a technology company that supplies semiconductor and infrastructure software solutions.
```

File: Broadcom/symantec-endpoint-protection/_meta/fields.yml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+threat.enrichments.indicator:
+  name: threat.enrichments.indicator
+  description: ""
+  type: dict
+
+threat.enrichments.indicator.type:
+  name: threat.enrichments.indicator.type
+  description: ""
+  type: text
```
Comment (PierrickV) [inline Broadcom/symantec-endpoint-protection/_meta/fields.yml:14]:
```suggestion
  type: keyword
```
In ECS, this field is a keyword

File: Broadcom/symantec-endpoint-protection/_meta/fields.yml [hunk 3]
```diff
@@ -0,0 +1,156 @@
+threat.enrichments:
+  name: threat.enrichments
+  description: ""
+  type: array
+
+threat.enrichments.indicator:
+  name: threat.enrichments.indicator
+  description: ""
+  type: dict
```
Comment (PierrickV) [inline Broadcom/symantec-endpoint-protection/_meta/fields.yml:9]:
```suggestion
  type: object
```

`object` in ECS (I am not sure if it works in Ingest)

File: Broadcom/symantec-endpoint-protection/_meta/fields.yml [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+threat.enrichments.indicator.sightings:
+  name: threat.enrichments.indicator.sightings
+  description: ""
+  type: number
+
+threat.enrichments.indicator.description:
+  name: threat.enrichments.indicator.description
+  description: ""
+  type: text
```
Comment (PierrickV) [inline Broadcom/symantec-endpoint-protection/_meta/fields.yml:39]:
```suggestion
  type: keyword
```

type in ecs

File: Broadcom/symantec-endpoint-protection/_meta/manifest.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+uuid: 419bd705-fa61-496c-94fa-28d6c1f2e2a8
+name: Symantec Endpoint Protection
+slug: symantec-endpoint-protection
+
+description: >-
+  Symantec Endpoint Protection is a client-server solution that protects laptops, desktops, and servers in your network against malware, risks, and vulnerabilities
```
Comment (PierrickV) [inline Broadcom/symantec-endpoint-protection/_meta/manifest.yml:6]:
```suggestion
  Symantec Endpoint Protection is a client-server solution that protects laptops, desktops, and servers in your network against malware, risks, and vulnerabilities.
```

File: Broadcom/symantec-endpoint-protection/_meta/smart-descriptions.json [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        },
+        {
+            "field": "file.path"
+        }
+    ],
+    "relationships": [
+        {
+            "source": "host.hostname",
+            "target": "file.path",
+            "type": "execute"
```
Comment (PierrickV) [inline Broadcom/symantec-endpoint-protection/_meta/smart-descriptions.json:22]:
```suggestion
            "type": "executed"
```

File: Broadcom/symantec-endpoint-protection/_meta/smart-descriptions.json [hunk 7]
```diff
@@ -0,0 +1,52 @@
+[{
+    "value": "Allowed execution of {file.path} on {host.hostname} ({host.ip})",
```
Comment (PierrickV) [inline Broadcom/symantec-endpoint-protection/_meta/smart-descriptions.json:2]: A sentence starting by "Allowed" does not feel right (who allowed ?)

File: Broadcom/symantec-endpoint-protection/_meta/smart-descriptions.json [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        },
+        {
+            "field": "event.action"
+        }
+    ],
+    "relationships": [
+        {
+            "source": "host.hostname",
+            "target": "file.path",
+            "type": "execute"
```
Comment (PierrickV) [inline Broadcom/symantec-endpoint-protection/_meta/smart-descriptions.json:49]:
```suggestion
            "type": "executed"
```

File: Broadcom/symantec-endpoint-protection/_meta/smart-descriptions.json [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        }
+    ]
+},{
+    "value": "Virus found on {host.hostname} ({host.ip}): {file.path} was {event.action}",
```
Comment (PierrickV) [inline Broadcom/symantec-endpoint-protection/_meta/smart-descriptions.json:26]:
```suggestion
    "value": "Malware detected on {host.hostname} ({host.ip}): {file.path} was {event.action}",
```

Seems more appropriate

Comment (PierrickV) [inline Broadcom/symantec-endpoint-protection/_meta/smart-descriptions.json:26]: I didn't get the end of the sentence `was {event.action}`.

I suggest that you modified this smart description by using the following sentence

```
    "broadcom": {
      "endpoint_protection": {
        "source": "Auto-Protect scan",
```

**identified**
```
    "file": {
      "path": "C:\\Users\\admin\\Desktop\\test.txt",
```

**as**
```
    "rule": {
      "name": "Virus",
```

**and applied the action**

```
  "broadcom": {
        "action": {
          "main": "Cleaned by deletion",
```

File: Broadcom/symantec-endpoint-protection/_meta/smart-descriptions.json (hunk 8 above)
Comment (PierrickV) [reply Broadcom/symantec-endpoint-protection/_meta/smart-descriptions.json:49]: was the file really executed or it is like a file-system scan that detected a malicious file ?

**Review by PierrickV — APPROVED** [review]
:rocket:

**Pull request merged**